lib/scarpe/stack.rb

```diff
@@ -14,10 +18,57 @@ def initialize(width: nil, height: nil, margin: nil, padding: nil, scroll: false
 
       super
 
-      # Create the display-side widget *before* instance_eval, which will add child widgets with their display widgets
       create_display_widget
+      # Create the display-side widget *before* instance_eval, which will add child widgets with their display widgets
 
       instance_eval(&block)
     end
   end
+
+  class Widget
+    def download(url)
+      Thread.new do
+        begin
+          uri = URI(url)
+          http = Net::HTTP.new(uri.host, uri.port)
+          http.use_ssl = true
+          http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+
+          request = Net::HTTP::Get.new(uri.request_uri)
+
+          http.request(request) do |response|
+            case response
+            when Net::HTTPSuccess
+              content = response.body
+
+              puts "Content is:\n#{content}"
```

Comment:
Do the puts statements here want to be removed? I would assume so.

Reply:
will remove